network/network.go

```diff
@@ -219,687 +172,303 @@ type network struct {
-		peerListGossipFreq:           peerListGossipFreq,
-		peerListGossipSize:           peerListGossipSize,
-		peerListStakerGossipFraction: peerListStakerGossipFraction,
-		getVersionTimeout:            getVersionTimeout,
-		allowPrivateIPs:              allowPrivateIPs,
-		gossipAcceptedFrontierSize:   gossipAcceptedFrontierSize,
-		gossipOnAcceptSize:           gossipOnAcceptSize,
-		pingPongTimeout:              pingPongTimeout,
-		pingFrequency:                pingFrequency,
-		disconnectedIPs:              make(map[string]struct{}),
-		connectedIPs:                 make(map[string]struct{}),
-		peerAliasIPs:                 make(map[string]struct{}),
-		peerAliasTimeout:             peerAliasTimeout,
-		retryDelay:                   make(map[string]time.Duration),
-		myIPs:                        map[string]struct{}{ip.IP().String(): {}},
-		readBufferSize:               readBufferSize,
-		readHandshakeTimeout:         readHandshakeTimeout,
-		inboundConnThrottler:         throttling.NewInboundConnThrottler(log, inboundConnThrottlerConfig),
-		healthConfig:                 healthConfig,
-		benchlistManager:             benchlistManager,
-		tlsKey:                       tlsKey,
-		latestPeerIP:                 make(map[ids.ShortID]signedPeerIP),
-		byteSlicePool: sync.Pool{
-			New: func() interface{} {
-				return make([]byte, 0, defaultByteSliceCap)
-			},
-		},
-		compressionEnabled:   compressionEnabled,
-		inboundMsgThrottler:  inboundMsgThrottler,
-		outboundMsgThrottler: outboundMsgThrottler,
-		whitelistedSubnets:   whitelistedSubnets,
+		log:                         log,
+		currentIP:                   config.MyIP,
+		parser:                      version.NewDefaultApplicationParser(),
+		listener:                    listener,
+		router:                      router,
+		dummyNodeID:                 rand.Uint32(),
+		disconnectedIPs:             make(map[string]struct{}),
+		connectedIPs:                make(map[string]struct{}),
+		peerAliasIPs:                make(map[string]struct{}),
+		retryDelay:                  make(map[string]time.Duration),
+		myIPs:                       map[string]struct{}{config.MyIP.IP().String(): {}},
+		inboundConnUpgradeThrottler: throttling.NewInboundConnUpgradeThrottler(log, config.ThrottlerConfig.InboundConnUpgradeThrottlerConfig),
+		benchlistManager:            benchlistManager,
+		latestPeerIP:                make(map[ids.ShortID]signedPeerIP),
+		versionCompatibility:        version.GetCompatibility(config.NetworkID),
+		config:                      config,
+		mc:                          msgCreator,
+	}
+
+	netw.serverUpgrader = NewTLSServerUpgrader(config.TLSConfig)
+	netw.clientUpgrader = NewTLSClientUpgrader(config.TLSConfig)
+
+	netw.dialer = dialer.NewDialer(constants.NetworkType, config.DialerConfig, log)
+	primaryNetworkValidators, ok := config.Validators.GetValidators(constants.PrimaryNetworkID)
+	if !ok {
+		return nil, errNoPrimaryValidators
+	}
```

Comment:
This could present a problem, as we might not have validators here yet, thus crashing the network start-up.